### DISCUSSION TITLE
unable to load admin dashboard

### DISCUSSION THREAD

**Discussion by saleemhaq** [OrchardCMS/OrchardCore#12117] [Q&A]
When i try to login to orchard dashboard go this below error:

![image](https://user-images.githubusercontent.com/5202291/181845279-f90af069-7881-4073-816c-b1a625c2dd3f.png)

Actually I created new role like subadmin and add new user and assign this role to user and then we got above error, please advise

**Comment by ns8482e** [comment]
Clear you cookies

**Reply by saleemhaq** [reply]
> Clear you cookies

still facing same when try this with 3 different systems also in server but got the same results?

**Reply by saleemhaq** [reply]
> Clear you cookies

browser incognito window i open the same but got the same results :(

**Reply by hishamco** [reply]
Are you loading a BIG data into your admin dashboard?

**Reply by saleemhaq** [reply]
> Clear you cookies

can you please advise me how to do to resolve this problem in one go ?? please help @ns8482e

**Comment by saleemhaq** [comment]
Yes all the rights that we have in admin exclude of delete see below: 

![image](https://user-images.githubusercontent.com/5202291/181993116-79701da6-9792-4e47-b5c0-fa5f2004ac99.png)

that we give to other role so what we do please suggest.

**Reply by hishamco** [reply]
You didn't answer my eailer question?!!

**Reply by saleemhaq** [reply]
I not load much data but mid level of data as i load on admin it work as earlier i tell you the step of reproduce is that i create new role and assign the rights to them so the application is behave like that.

**Reply by saleemhaq** [reply]
> You didn't answer my eailer question?!!

If want to load big data so what kind of things need to do please advise?

**Reply by hishamco** [reply]
Just try to comment the loaded and try. May be there are much much data have been send in HTTP header

**Comment by ns8482e** [comment]
@saleemhaq the issue might be that your header is large and may be due to having more instances running on same host/port localhost:5001 each has saved Auth cookies

As each instance set’s their own Auth cookie, but running on same host/port  browser sends headers for all resulting  header being large


I’d suggest if you are running more than one OC instance on same dev machine then run each on different port

**Reply by saleemhaq** [reply]
> @saleemhaq the issue might be that your header is large and may be due to having more instances running on same host/port localhost:5001 each has saved Auth cookies
> 
> As each instance set’s their own Auth cookie, but running on same host/port browser sends headers for all resulting header being large
> 
> I’d suggest if you are running more than one OC instance on same dev machine then run each on different port

When i try to login other role of user see cookie:

![image](https://user-images.githubusercontent.com/5202291/182162795-6e2ffd64-a3d6-48e6-a555-7fee4473236b.png)

but when login with admin role there is a difference in cookie even i not change any things 

![image](https://user-images.githubusercontent.com/5202291/182163142-52f87e75-aa1f-4d33-ae68-f21bfb3bbeca.png)

so admin is loaded but other role not @ns8482e @hishamco  please help

**Reply by ns8482e** [reply]
Not sure why do you have six chunks of cookie- Try deleting all six chunks and and cookie and try login again see if that works

**Reply by saleemhaq** [reply]
> Not sure why do you have six chunks of cookie- Try deleting all six chunks and and cookie and try login again see if that works

yes i delete all and relogin with that user still facing the same?? @ns8482e 

see when login:

![image](https://user-images.githubusercontent.com/5202291/182165410-77323007-7d6d-4fc2-bb74-a19a52d35aee.png)

after login with that user:

![image](https://user-images.githubusercontent.com/5202291/182165495-dfd4091f-6b20-4953-8d5f-cd500875ba14.png)

please help @ns8482e

**Reply by ns8482e** [reply]
Are you adding any custom claims?

**Reply by hishamco** [reply]
Something strange going on, of course you have a custom login that adding multiple AUTH cookies

**Reply by saleemhaq** [reply]
> Are you adding any custom claims?

No, i not add anything

**Reply by saleemhaq** [reply]
> Something strange going on, of course you have a custom login that adding multiple AUTH cookies

I not add any custom login interface just create new role first then create new user and assign that role and try to login with that user it is total re produce step

**Comment by ns8482e** [comment]
@saleemhaq It seems cookie size is going over default limit by server 

you can increase the limit on kestrel using following in Program.cs

`builder.WebHost.ConfigureKestrel(o=> o.Limits.MaxRequestHeadersTotalSize = 327680 );`

**Reply by saleemhaq** [reply]
> @saleemhaq It seems cookie size is going over default limit by server
> 
> you can increase the limit on kestrel using following in Program.cs
> 
> `builder.WebHost.ConfigureKestrel(o=> o.Limits.MaxRequestHeadersTotalSize = 327680 );`

i using IIS

**Reply by ns8482e** [reply]
Yes IIS has 16kb limit, you need to increase the limit in IIS then

**Reply by saleemhaq** [reply]
> Yes IIS has 16kb limit, you need to increase the limit in IIS then

@ns8482e can you help me out in that how to increase I decouple cms of .NET5 with dotnet core 3.1 I do the setting like that in web.config 

![image](https://user-images.githubusercontent.com/5202291/182692924-42c262b2-8dd1-4bab-84f3-4e9d10b99ac5.png)

but no luck still facing the same issue: @hishamco please help

**Reply by hishamco** [reply]
Can you clear browser data

**Reply by saleemhaq** [reply]
> Can you clear browser data

yes i do, but again i login but got the same issue issue please help @hishamco

**Comment by saleemhaq** [comment]
Can u guide me how I do that in proper way. To execute my request by that
user.

**Reply by saleemhaq** [reply]
> Can u guide me how I do that in proper way. To execute my request by that user.

Can u guide me how I do that in proper way. To execute my request by that
user. @ns8482e

**Comment by ns8482e** [comment]
IIS is tough it uses registry, something you need to search in google -

**Reply by saleemhaq** [reply]
Yes I search it out and try to set in registry but still got the same issue:( @ns8482e

**Reply by ns8482e** [reply]
If you can’t increase limit on your web server then you may need to save claims on server side session instead of cookie by implementing  custom `ITicketStore` 

Currently OC doesn’t have that feature - see issue created #12142

**Reply by saleemhaq** [reply]
> If you can’t increase limit on your web server then you may need to save claims on server side session instead of cookie by implementing custom `ITicketStore`
> 
> Currently OC doesn’t have that feature - see issue created #12142

Can you help me @ns8482e  in that feature because from ten days i suffer for this feature can this module owner help us also, Folks please help @hishamco

**Reply by ns8482e** [reply]
@saleemhaq sure I’ll help and contribute when time permits

**Comment by infofromca** [comment]
@saleemhaq you can implement it by taking #9834 as a ref.

**Comment by hyzx86** [comment]
I have encountered similar problems before. I used openid authorization + JWT. I used spa program to re implement the role permission management interface of OC. However, I incorrectly checked all permissions, but because all permissions are encoded in the JWT token

This token will be sent to OC through request.headers.Authorization.
This problem occurs

**Comment by saleemhaq** [comment]
Thanks.

**Comment by saleemhaq** [comment]
So what can I do? Please let me know I suffer from last 10 days what I do?

**Reply by ns8482e** [reply]
PR open check it out
